### PR TITLE
Plot testing fixes

### PR DESCRIPTION
**Pull request by sharlagelfand**
Fixes for plotting and especially for testing plots:

* Argument added to `magora_boxplot()` to use/not use `theme_sage()`, with using as default within the app (and not using reserved for testing)
* "No data" annotation font updated to actually use Lato (except in the case when `use_theme_sage = FALSE`)
* Fixed mock testing data that was not treating `sex` as a factor, causing NA `sex` to appear in the legend of some test plots

Once this is confirmed good and merged I'll rebase in #22 and update the tests there so those pass too

### CODE REVIEW

File: R/magora_boxplot.R [hunk 1]
```diff
@@ -50,17 +51,23 @@ magora_boxplot <- function(data, plot_type = c("phenotype", "gene expression"))
...
     ggplot2::labs(x = "Age", y = switch(plot_type,
       "phenotype" = unique(data[["phenotype"]]),
       "gene expression" = "Transcripts Per Million (TPM)"
     )) +
-    sagethemes::scale_fill_sage_d() +
-    sagethemes::theme_sage(base_size = 16) +
+    sagethemes::scale_fill_sage_d()
+
+  if (use_theme_sage) {
+    p +
```
Comment (karawoo) [inline R/magora_boxplot.R:66]:
```suggestion
    p <- p +
```

**Review by karawoo — CHANGES_REQUESTED** [review]
Thanks for sorting this out! I think there's one change needed to make sure the sage theme gets added. 

If it's easier, instead of the `use_theme_sage` argument you could keep `magora_boxplot()` as it was and add `+ ggplot2::theme_test()` to the tests, which should override the sage theme there. The new argument works too though 👍

File: R/magora_boxplot.R (hunk 1 above)
Comment (sharlagelfand) [reply R/magora_boxplot.R]: Ah thank you!

**Comment by sharlagelfand** [comment]
Thanks for catching that! I'd prefer to use the new argument since it handles the `geom_text()` font easily too - doesn't look like `theme_test()` changes it back to the default

**Review by karawoo — APPROVED** [review]
Makes sense! I'll wait for the builds to pass before merging, but this looks good to me.

**Pull request merged**